MQTTPacket/test/test2.c

```diff
@@ -400,11 +400,13 @@ int test1(struct Options options)
 	data.will.qos = 1;
 	data.will.retained = 0;
 	data.will.topicName.cstring = "will topic";
+	data.will.properties = &willProperties;
 
-	rc = MQTTV5Serialize_connect(buf, buflen, &data, &connectProperties, &willProperties);
+	rc = MQTTV5Serialize_connect(buf, buflen, &data, &connectProperties);
 	assert("good rc from serialize connect", rc > 0, "rc was %d\n", rc);
 
-	rc = MQTTV5Deserialize_connect(&outWillProperties, &outConnectProperties, &data_after, buf, buflen);
+	data_after.will.properties = &outWillProperties;
```

Comment:
This is debatable: in previous cases we wouldn't ask for the application to provide a value for out-data. Now this is required in order to provide the memory allocation for properties.